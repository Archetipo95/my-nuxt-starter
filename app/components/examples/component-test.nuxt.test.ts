import { renderSuspended } from "@nuxt/test-utils/runtime"
import { describe, expect, it } from "vitest"

import type { ComponentTestProps } from "./component-test.vue"

import ComponentTest from "./component-test.vue"

const messageValid: ComponentTestProps = {
  state: "valid",
}

const messageInvalid: ComponentTestProps = {
  state: "invalid",
}

describe("atomsMessage.vue", () => {
  describe("check the output of our component in different given states", async () => {
    it.each([
      ["stateValid", messageValid],
      ["stateInvalid", messageInvalid],
    ])("renders %s correctly", async (_, props: ComponentTestProps) => {
      const {
        unmount,
        html,
      } = await renderSuspended(ComponentTest, {
        props,
        slots: {
          title: () => `State message: ${props.state}`,
        },
      })

      expect(html()).toMatchSnapshot()

      unmount()
    })
  })
})
