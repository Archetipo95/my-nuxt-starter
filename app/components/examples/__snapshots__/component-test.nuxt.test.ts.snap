// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`atomsMessage.vue > check the output of our component in different given states > renders stateInvalid correctly 1`] = `
"<div id="test-wrapper">
  <div class="text-red-500">
    <div class="heading">State message: invalid</div>
  </div>
</div>"
`;

exports[`atomsMessage.vue > check the output of our component in different given states > renders stateValid correctly 1`] = `
"<div id="test-wrapper">
  <div class="text-green-500">
    <div class="heading">State message: valid</div>
  </div>
</div>"
`;
